src/atdf/interrupt.rs

```diff
@@ -0,0 +1,26 @@
+use crate::{
+    chip,
+    chip::Interrupt,
+    util,
+    ElementExt,
+};
+
+pub fn parse_list(ints: &xmltree::Element) -> crate::Result<Vec<chip::Interrupt>> {
+    let mut interrupts = vec![];
```

Comment:
IMO, it would be even better to make the function only parse a single interrupt and move out the iteration into `chip.rs`.

```rust
// chip.rs
let interrupts = device
	.first_child("interrupts")?
	.children
	.iter()
	.map(atdf::interrupt::parse)
	.map(|r| r.map(|i| (i.name.clone(), i)))
	.collect::<Result<_, _>>()?;

// interrupt.rs
pub fn parse(el: &xmltree::Element) -> crate::Result<chip::Interrupt> {
}
```